_search-plugins/neural-search.md

```diff
@@ -1,176 +1,203 @@
+To perform vector search on your index, use the `neural` query clause either in the [k-NN plugin API]({{site.url}}{{site.baseurl}}/search-plugins/knn/api/#search-model) or [Query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/index/) queries. You can refine the results by using a [k-NN search filter]({{site.url}}{{site.baseurl}}/search-plugins/knn/filter-search-knn/).
 
-### Neural request fields
+### Neural query request fields
 
-Include the following request fields under the `neural` field in your query:
+Include the following request fields under the `neural` query clause:
+
+```json
+"neural": {
+  "<vector_field>": {
+    "query_text": "<query_text>",
+    "model_id": "<model_id>",
+    "k": 100
+  }
+}
+```
+
+The top-level `vector_field` specifies the vector field against which to run a search query. The following table lists the other neural query fields.
 
 Field | Data type | Description
 :--- | :--- | :--- 
-vector_field | string | The vector field against which to run a search query.
-query_text | string | The query text from which to produce queries.
-model_id | string | The ID of the model that will be used in the embedding interface. The model must be indexed in OpenSearch before it can be used in Neural Search.
-k | integer | The number of results the k-NN search returns.
+`query_text` | String | The query text from which to generate text embeddings.
+`model_id` | String | The ID of the model that will be used to generate text embeddings from the query text. The model must be indexed in OpenSearch before it can be used in neural search.
+`k` | Integer | The number of results the k-NN search returns.
 
 ### Example request
 
-The following example request uses a search query that returns vectors for the "Hello World" query text:
-
+The following example request uses a Boolean query to combine a filter clause and two query clauses---a neural query and a `match` query. The `script_score` query assigns custom weights to the query clauses:
 
 ```json
-GET my_index/_search
+GET /my-nlp-index/_search
 {
+    "_source": {
+    "excludes": [
+      "passage_embedding"
+    ]
+  },
   "query": {
-    "bool" : {
-      "filter": {
-        "range": {
-          "distance": { "lte" : 20 }
-        }
+    "bool": {
+    "filter": {
+         "wildcard":  { "id": "*1" }
       },
-      "should" : [
+      "should": [
         {
           "script_score": {
             "query": {
               "neural": {
-                "passage_vector": {
-                  "query_text": "Hello world",
-                  "model_id": "xzy76xswsd",
+                "passage_embedding": {
+                  "query_text": "Hi world",
```

Comment:
I believe here "Hi planet" would come.

Reply:
Actually, I didn't want to search for the exact text here because vector search would match both documents anyway. Do you think it makes more sense to search for the exact text?

Reply:
I think we should always give user a sucessful case so he goes in right direction

Reply:
Both GET requests return results. I have added the responses to show the results.